src/content/developers/docs/consensus-mechanisms/pos/index.md

```diff
@@ -29,6 +29,18 @@ To participate as a validator, a user must deposit 32 ETH into the deposit contr
+2. The wallet sends the transaction to an Ethereum [execution client](/developers/docs/nodes-and-clients/) which verifies its validity.
+3. If the transaction is valid, the execution client adds it to its mempool and broadcasts it to other nodes, making it part of the public mempool. Advanced users may have sent their transaction instead to specialized and extended execution clients that hide the transaction from the public mempool, making it available to specialized block builders such as [Flashbots Auction](https://docs.flashbots.net/flashbots-auction/overview).
+4. The validator client, that has randomly been chosen as the block proposer for the current slot, proposes the next block that defines the new state for the network and broadcasts it to the network. The block proposer can freely choose which valid transactions to include from their local mempool into the block, or can ask an external block builder through a middleware service such as [MEV-Boost](https://docs.flashbots.net/flashbots-mev-boost/introduction) to build a block for them. The communication between the execution and consensus clients is described in more detail in [Connecting the Consensus and Execution Clients](/developers/docs/networking-layer/#connecting-clients).
+5. The consensus clients from other validators on the network track the newly proposed block and verify it via their execution client. The validators then publicly attest as to whether the block is valid or not. If they accept it, their consensus client adds it to the head to their version of the chain and attest to the current state of the chain (once an epoch).
+6. The other consensus clients on the network run a [fork selection algorithm](/developers/docs/consensus-mechanisms/pos/#fork-choice) and add the block to the fork containing blocks with the most attestations from all validators (weighted by staked Ether).
+7. The transaction can be considered to have "finality", i.e., that it can not be reverted, when the block that it's been included in has become part of a chain with a subsequent finalized checkpoint block. This occurs, under normal circumstances (i.e., if no major validators go offline or major client bugs have occurred) after two epochs, or approximately 12.8 minutes. Although typically, the chain, and therefore our user's transaction, can be assumed as having reached finality with high probability much sooner in practice. This can be assessed by monitoring the attestation data contained in previous the blocks' metadata. More detail on finality can be found in the [next section](/developers/docs/consensus-mechanisms/pos/#finality).
```

Comment:
```suggestion
6. The transaction can be considered "finalized", i.e., that it can not be reverted, if it has become part of a chain with a "supermajority link" between two checkpoints. Checkpoints occur at the start of each epoch and to have a supermajority link they must both be attested to by 66% of the total staked ETH on the network.

More detail on finality can be found below.
```